source/planner.js

```diff
@@ -10,7 +10,185 @@ function init(){
+        <textarea id="new-task-description" placeholder="Task Description"></textarea>
+        <div class="tags-container">
+            <div class="tag">Tags</div>
+            <div class="tag">Tags</div>
+            <div class="tag">Tags</div>
+            <button class="add-tag-button">+</button>
+        </div>
+        <input type="date" id="new-task-date">
+        <input type="time" id="new-task-time">
+        <div class="popup-buttons">
+            <button class="popup-button-confirm" id="add-task-confirm">Add Task</button>
+            <button class="popup-button-cancel" id="add-task-cancel">Cancel</button>
+        </div>
+    </div>
+`   );
+
+    // Event listener for opening the add task popup
+    addTaskButton.addEventListener('click', () => {
+        addTaskPopup.style.display = 'block';
+        overlay.style.display = 'block';
+    });
+
+    // Event listener for closing the add task popup
+    addTaskCancel.addEventListener('click', () => {
+        addTaskPopup.style.display = 'none';
+        overlay.style.display = 'none';
+    });
+
+    // Event listener for adding a new task
+    addTaskConfirm.addEventListener('click', () => {
+        const title = document.getElementById('new-task-title').value;
+        const description = document.getElementById('new-task-description').value;
+        const date = new Date(document.getElementById('new-task-date').value);
+        const time = convertTo12Hour(document.getElementById('new-task-time').value);
+        if (title && description && date) {
+            addNewTask(title, description, date, time);
+            addTaskPopup.style.display = 'none';
+            overlay.style.display = 'none';
+        }
+    });
+
+    /**
+     * Adds a new task to the task list and the calendar.
+     * @param {string} title - The title of the task.
+     * @param {string} description - The description of the task.
+     * @param {Date} date - The due date of the task.
+     * @param {string} time - The due time of the task.
+     */
+    function addNewTask(title, description, date, time) {
+        const taskList = document.querySelector('.task-list ul');
+        const calendarDays = document.querySelectorAll('.day');
+    
+        const taskItem = document.createElement('li');
+        taskItem.className = 'task-item';
+    
+        const taskTitle = document.createElement('div');
+        taskTitle.className = 'task-title';
+        taskTitle.textContent = title;
+    
+        const taskDescription = document.createElement('div');
+        taskDescription.className = 'task-description';
+        taskDescription.textContent = description;
+    
+        taskItem.appendChild(taskTitle);
+        taskItem.appendChild(taskDescription);
+        taskList.appendChild(taskItem);
+    
+        calendarDays.forEach(day => {
+            if (day.id === date.toDateString()) {
+                const eventItem = document.createElement('li');
```

Comment:
maybe its better to create a helper function for this or use a custom element since this code requires so many different html to be made it will weigh down performance